pkg/stream/http2/stream.go

```diff
@@ -435,13 +436,16 @@ type serverStream struct {
 func (s *serverStream) AppendHeaders(ctx context.Context, headers api.HeaderMap, endStream bool) error {
 	var rsp *http.Response
 
+	// --- do not pass internal headers to upstream
 	var status int
 	if value, _ := headers.Get(types.HeaderStatus); value != "" {
 		headers.Del(types.HeaderStatus)
 		status, _ = strconv.Atoi(value)
 	} else {
 		status = 200
 	}
+	headers.Del(types.HeaderScheme)
```

Comment:
It should be call it in the function of client's [AppendHeaders](https://github.com/mosn/mosn/pull/1261/files#diff-70118d13dd7afdc9f6b24253246ae52fR805) rather than serverStream.